tmt/steps/provision/__init__.py

```diff
@@ -1268,6 +1276,25 @@ def is_ready(self) -> bool:
         # Enough for now, ssh connection can be created later
         return self.guest is not None
 
+    def setup(self) -> None:
+        if self.is_dry_run:
+            return
+        if self.guest is None and not self.is_dry_run:
+            raise tmt.utils.GeneralError('The guest is not available.')
+        if not self.facts.is_superuser:
+            parent = cast(Provision, self.parent)
+            ssh_command: tmt.utils.Command = self._ssh_command()
+
+            remote_commands: ShellScript = ShellScript(f'sudo dnf install -y acl')
+            workdir_root = effective_workdir_root()
+            remote_commands += ShellScript(f'setfacl -d -m o:rX {workdir_root}')
+            ssh_command += [
+                self._ssh_guest(),
+                remote_commands.to_element()
+                ]
+            self._run_guest_command(ssh_command, silent=True)
```

Comment:
I don't understand, is `guest.execute()` unusable for some reason?

Reply:
What I had in mind was that `execute` is a function the class makes available to other client code, and it is usually invoked after the guest is initialized. Basically, the expected sequence of states in a guest would be: `started`, `setup`, `ready for executes`. Therefore, since the expected use of `execute` is after initialization, I can see a developer coming later on and modifying `execute` with the assumption the guest is already fully setup, but in reality is not, so creating a bug in `setup` (`setup` calls `execute`, but `execute` expected `setup` already invoked).

So I went head and broke the dependency of `setup` in `execute`, and instead made it use the lower lever function `_run_guest_command`. My thought behind this is that the semantic of these functions and the expectations they create will drive the developers decision on where to put a change later on.

It is probably simpler to just invoke `execute`, though. It helps that it is just right there in the same class so bugs out of wrong assumptions are easy to spot.